src/jit/codegenlegacy.cpp

```diff
@@ -5426,6 +5426,9 @@ void CodeGen::genCodeForTreeLeaf_GT_JMP(GenTreePtr tree)
             offset += REGSIZE_BYTES;
         }
     }
+
+    // Lock the argument registers s.t. they are not trashed by code that runs before the jmp (e.g. GS cookie checks).
+    regSet.rsLockReg(fixedArgsMask);
```

Comment:
@BruceForstall JIT32's code is identical to the legacy backend code before these changes. There isn't really the opportunity for unbounded codegen here: this is only done during the codegen for a `JMP` node, which must be a block terminator. AFAIK the only code that may be generated before the registers are unlocked is the code for a GS cookie check.